Use MultiMap set method instead of add for client assertion parameters

Sopka opened the pull request:
The issue is in OidcClientImpl.java, where `add()` is used instead of `set()` for JWT bearer authentication parameters. Other authentication methods correctly use `set()`.

This causes a mismatch between the expected behavior (replacing previous values) and the actual behavior (accumulating values).

Replaced the `add()` method calls with `set()` in the JWT bearer authentication code path:

```java
// Replace these lines:
body.add(OidcConstants.CLIENT_ASSERTION, clientAssertion);
body.add(OidcConstants.CLIENT_ASSERTION_TYPE, OidcConstants.JWT_BEARER_CLIENT_ASSERTION_TYPE);

// With:
body.set(OidcConstants.CLIENT_ASSERTION, clientAssertion);
body.set(OidcConstants.CLIENT_ASSERTION_TYPE, OidcConstants.JWT_BEARER_CLIENT_ASSERTION_TYPE);
```

This would ensure only the newest key-value pairs are included in the request, eliminating the duplication problem.

- Fixes: #48879

Sopka committed:
Use MultiMap set method instead of add for client assertion parameters

sberyozkin commented:
Thanks @Sopka.  Indeed, would be good to have a test.

The test which sends a single request is [here](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/OidcClientTest.java#L51).
The wiremock stub is [here](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-client-wiremock/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L45).

I guess the simplest option is to do 2 calls in that test and tune the stub definition to fail if more than one client_assertion parameter is used. Or may be you can just update https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-client-wiremock/src/main/java/io/quarkus/it/keycloak/OidcRequestCustomizer.java to check if the body contains only one of those parameters.

Have a look please, I can help if you'd like

Sopka committed:
Test jwt bearer authentication with multiple posts to OIDC server

Ensure that all subsequent requests to the OIDC server's token endpoint include the correct number of form parameters in the POST data.

Sopka commented:
Added a test that makes multiple requests to the OIDC token endpoint using a filter configuration that always forces new token requests. This approach avoids issues with simulating token expiration and eliminates the need to wait for tokens to expire. The test verifies that each request sends the expected form parameters.

sberyozkin reviewed (approved):
Thanks @Sopka.

I think we can probably keep both commits, without squashing, as each of them is quite self-contained.
@geoand I'll merge soon if there are no objections, I can easily squash if preferred
